airgun/browser.py

```diff
@@ -196,10 +196,32 @@ def _get_selenium_browser(self):
             self._webdriver = webdriver.PhantomJS(
                 service_args=['--ignore-ssl-errors=true'])
         elif self.browser == 'remote':
-            capabilities = vars(settings.webdriver_desired_capabilities)
+            browserName = settings.webdriver_desired_capabilities.browserName
+            if browserName == 'chrome':
+                capabilities = webdriver.DesiredCapabilities.CHROME.copy()
+            elif browserName == 'firefox':
+                capabilities = webdriver.DesiredCapabilities.FIREFOX.copy()
+            elif browserName == 'ie':
+                capabilities = (
+                        webdriver.DesiredCapabilities.INTERNETEXPLORER.copy())
+            elif browserName == 'edge':
+                capabilities = webdriver.DesiredCapabilities.EDGE.copy()
+            else:
+                raise ValueError(
+                    '"{}" browserName is not supported. Please use one of {}'
+                    .format(browserName, ('chrome', 'firefox', 'ie', 'edge'))
+                )
+            if settings.webdriver_desired_capabilities:
+                capabilities.update(
+                    vars(settings.webdriver_desired_capabilities))
```

Comment:
Is this whole part (including `idle_timeout` below) so different from code in `_get_saucelabs_browser` that we should have it independent? Can't we extract it to separate private method and just run for both saucelab and remote? In fact, saucelab is just a special case of remote.

Also, we always set `download.prompt_for_download` to False and pass disable-web-security to Chrome, and set `acceptSslCerts` and `javascriptEnabled` to True for Edge. I think you want to do it in this path, too.

As a side note, this change shows that `SeleniumBrowserFactory` is difficult to extend and there is a lot of code duplication. It probably should be rewritten to use separate classes, so Sauce and Docker can simply inherit from Remote (and Remote should inherit from normal local webdriver, where all capabilities are set?). But refactoring is probably out of scope for this PR.

Reply:
Yes, I absolutely agree with you, the capabilities for different browser can be public, Remote/Sauce can use the same capabilities, I'd like to create a new PR for this.